husky-team/husky
# Add Redis Connector.

# Test cases

**- Preparation**

1. Flush Redis data using "/data/opt/course/redis/flushall-cluster.sh"
2. Check configuration file "$HUSKY_ROOT/test/test-redis-config.ini"

**- Test 1: CMake**

1. Follow the standard Cmake commands of Husky.
2. make "Master".
3. make "TestRedisOutputFormat".
4. make "TestRedisInputOutputFormat".
5. make "WordCountRedis".

**Expectation**: running well.
**Notes**: 

**- Test 2: enwiki from MongoDB to Redis:**

1. Run "./Master -C test-redis-config.ini"
2. Run "time /data/opt/tmp/exec.sh ./TestRedisOutputFormat -C test-redis-config.ini"

**Expectation**: running well. 4825150 records to be imported. around 1m30s.
**Notes**: **MUST wait for 10 minutes before Test 3,** because for the first time data were loaded into Redis cluster, it takes time for master-slave synchronization.

**- Test 3: enwiki from Redis to Redis:**

1. Run "./Master -C test-redis-config.ini"
2. Run "time /data/opt/tmp/exec.sh ./TestRedisInputOutputFormat -C test-redis-config.ini"

**Expectation**: running well. 4825150 records to be imported. around 1m.
**Notes**: On Husky-worker side, "waiting for keys" logs may be printed as expected, which means that some workers are waiting for keys from the next batch.

**- Test 4: enwiki WordCount:**

1. Run "./Master -C test-redis-config.ini"
2. Run "time /data/opt/tmp/exec.sh ./WordCountRedis -C test-redis-config.ini"

**Expectation**: running well. 4825150 records to be imported. around 50s.
**Notes**: Can be performed consecutively without restart Husky-Master.
